components/edit_channel_header_modal/edit_channel_header_modal.jsx

```diff
@@ -46,50 +34,37 @@ class EditChannelHeaderModal extends React.PureComponent {
          */
         ctrlSend: PropTypes.bool.isRequired,
 
-        /*
-         * object with info about server error
-         */
-        serverError: PropTypes.object,
-
-        /*
-         * string with info about about request
-         */
-        requestStatus: PropTypes.string.isRequired,
-
         /*
          * Collection of redux actions
          */
         actions: PropTypes.shape({
 
+            closeModal: PropTypes.func.isRequired,
+
             /*
              * patch channel redux-action
              */
             patchChannel: PropTypes.func.isRequired,
         }).isRequired,
     }
 
-    static getDerivedStateFromProps(props) {
-        const {requestStatus} = props;
-        if (requestStatus === RequestStatus.SUCCESS) {
-            return {show: false};
-        }
-
-        return null;
-    }
-
     constructor(props) {
         super(props);
 
         this.state = {
             preview: false,
             header: props.channel.header,
-            show: true,
+            saving: false,
         };
     }
 
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
```

Comment:
Move `intl` into context since I'm not able to initiate some tests when it's on props.

Reply:
I hate to say it, but you'll have to revert this. There's an outstanding PR to upgrade react-intl to v3 which doesn't support the old context API.

For the tests, is the problem that you can't call instance methods? If so, perhaps `injectIntl` has a method like the `getWrappedInstance` on components wrapped with `connect`

Reply:
Thanks.  I used `dive` and it works.